### ISSUE TITLE
Code Style

### ISSUE THREAD

**Issue by VoyTechnology** [DethRaid/nova-renderer#5]
What coding style is used for this project? I know it might spawn a debate but setting one standard would help in code management.

There are a bunch that can be followed, for example [Google's](https://google.github.io/styleguide/cppguide.html) or [CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md)

**Comment by DethRaid** [comment]
I'm not importing a coding style wholesale. The [Coding Style wiki page](https://github.com/DethRaid/nova-renderer/wiki/Code-Style-Guidelines) is a collection of some of my habits that I think improve code readability.

Cpp Core Guidelines is definitely nice, but it's still in its early stages (Some sections still have ??? for the rationale) and it's rather large. I'd want to amend it a good bit, mostly for code formatting things, then people would have to read through the hundreds of rules in Cpp Core Guidelines and my amendments to get started, and that's a lot to read through.

I haven't read through Google's, so I don't really have an opinion on it, other than that it is also kinda large.

Ultimately, I don't want to choke development down with a lot of style rules. I trust that people who contribute to this project are pretty good coders and know enough to not write horrible code. I wanted to address a couple common issues, like tabs vs. spaces, in the Coding Style wiki page. If the need for more guidelines arises I'll definitely add them, but for now I don't really see a need for a large, comprehensive style guide.

*TL;DR:* We only have a couple style guidelines, and they're just some simple things to improve code readability and address common issues.
